### PR TITLE
Add a note for JDBC streaming result

### PR DESCRIPTION
**Pull request by 7yyo**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add a note for JDBC streaming result.

Case:
https://github.com/7yyo/jdbc-best-practices/blob/master/src/main/java/example/MysqlIO_41.java

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@7yyo Please involve a technical review. Thanks~

**Review by jackysp — APPROVED** [review]
LGTM

**Comment by 7yyo** [comment]
@TomShawn cc

File: best-practices/java-app-best-practices.md [hunk 1]
```diff
@@ -56,6 +56,10 @@ Java 应用尽管可以选择在不同的框架中封装，但在最底层一般
 在 JDBC 中通常有以下两种处理方式：
 
 - 设置 [`FetchSize` 为 `Integer.MIN_VALUE`](https://dev.mysql.com/doc/connector-j/5.1/en/connector-j-reference-implementation-notes.html#ResultSet) 让客户端不缓存，客户端通过 StreamingResult 的方式从网络连接上流式读取执行结果。
+- > 注意  
+  > 在使用流式读取数据时，需要将 resultset 读取完成或 close 后，才能继续使用该 stmt 进行下次查询，否则会报错 `No statements may be issued when any streaming result sets are open and in use on a given connection. Ensure that you have called .close() on any active streaming result sets before attempting more queries.`
+  > 如果需要在 resultset 读取完成或 close 前进行查询避免上述报错，可在 url 添加配置参数 `clobberStreamingResults=true`，这样会自动 close resultset，但之前流式查询未被读取的结果集会丢失。
```
Comment (7yyo) [inline best-practices/java-app-best-practices.md:61]: clobberStreamingResults

Comment (TomShawn) [inline best-practices/java-app-best-practices.md:61]:
```suggestion

    使用流式读取数据时，需要将 `resultset` 读取完成或 close 后，才能继续使用该语句进行下次查询，否则会报错 `No statements may be issued when any streaming result sets are open and in use on a given connection. Ensure that you have called .close() on any active streaming result sets before attempting more queries.`。
    
    如果需要在 `resultset` 读取完成或 close 前进行查询避免上述报错，可在 URL 中添加配置参数 `clobberStreamingResults=true`，这样会自动 close `resultset`，但之前流式查询未被读取的结果集会丢失。
```

**Review by TomShawn — APPROVED** [review]
LGTM

File: best-practices/java-app-best-practices.md [hunk 2]
```diff
@@ -56,6 +56,11 @@ Java 应用尽管可以选择在不同的框架中封装，但在最底层一般
 在 JDBC 中通常有以下两种处理方式：
 
 - 设置 [`FetchSize` 为 `Integer.MIN_VALUE`](https://dev.mysql.com/doc/connector-j/5.1/en/connector-j-reference-implementation-notes.html#ResultSet) 让客户端不缓存，客户端通过 StreamingResult 的方式从网络连接上流式读取执行结果。
+
+    使用流式读取数据时，需要将 `resultset` 读取完成或 close 后，才能继续使用该语句进行下次查询，否则会报错 `No statements may be issued when any streaming result sets are open and in use on a given connection. Ensure that you have called .close() on any active streaming result sets before attempting more queries.`。
+    
+    如果需要在 `resultset` 读取完成或 close 前进行查询避免上述报错，可在 URL 中添加配置参数 `clobberStreamingResults=true`，这样会自动 close `resultset`，但之前流式查询未被读取的结果集会丢失。
```
Comment (TomShawn) [inline best-practices/java-app-best-practices.md:62]:
```suggestion
    如果需要在 `resultset` 读取完成或 close 前进行查询避免上述报错，可在 URL 中添加配置参数 `clobberStreamingResults=true`，这样会自动 close `resultset`，但之前流式查询未被读取的结果集会丢失。 
```

**Pull request merged**